libs/blocks/iframe/iframe.js

```diff
@@ -1,13 +1,57 @@
 import { createTag } from '../../utils/utils.js';
 
+const ALLOWED_MESSAGE_ORIGINS = [
+  'https://stage.plan.adobe.com',
+  'https://plan.adobe.com',
+];
+
+function handleManagePlanEvents(message) {
+  const { subType, data } = message;
+  switch (subType) {
+    case 'EXTERNAL':
+      if (!data?.externalUrl || !data?.target) return;
+      window.open(data.externalUrl, data.target);
+      break;
+    case 'SWITCH':
+      if (!data?.externalUrl || !data?.target) return;
+      window.open(data.externalUrl, data.target);
+      break;
+    case 'Close':
+      document.querySelector('.dialog-modal')?.dispatchEvent(new Event('iframe:modal:closed'));
+      break;
+    default:
+      break;
+  }
+}
+
+export function handleIFrameEvents({ data }) {
+  let parsedMsg = null;
+  try {
+    parsedMsg = JSON.parse(data);
+  } catch (error) {
+    return;
+  }
+  const { app } = parsedMsg;
+  if (app === 'ManagePlan') handleManagePlanEvents(parsedMsg);
+}
```

Comment:
Nit: let's move everything into the try block
```suggestion
export function handleIFrameEvents({ data }) {
  try {
    const parsedMsg = JSON.parse(data);
    if (parsedMsg.app === 'ManagePlan') handleManagePlanEvents(parsedMsg);
  } catch (error) {
    return;
  }
}
```